execution/model/operator.go

```diff
@@ -5,10 +5,41 @@ package model
 
 import (
 	"context"
+	"time"
 
 	"github.com/prometheus/prometheus/model/labels"
 )
 
+type NoopTimingInformation struct{}
+
+func (ti *NoopTimingInformation) AddCPUTimeTaken(t time.Duration) {}
+
+type TimingInformation struct {
```

Comment:
I think in the future we will be using this struct for not just capturing CPU timing information so maybe it would be worth changing names in this PR before merging?

I suggest something like `TrackedTelemetry` and `NoopTelemetry`?